starknet_devnet/blueprints/rpc/structures/payloads.py

```diff
@@ -330,37 +352,92 @@ def rpc_fee_estimate(fee_estimate: FeeEstimationInfo) -> dict:
     return result
 
 
-def make_invoke_function(request_body: dict) -> InvokeFunction:
+def make_invoke_function(invoke_transaction: RpcBroadcastedInvokeTxn) -> InvokeFunction:
     """
-    Convert RPC request to internal InvokeFunction
+    Convert RpcBroadcastedInvokeTxn to InvokeFunction
     """
-    version = int(request_body.get("version", str(LEGACY_RPC_TX_VERSION)), 16)
-    nonce = request_body.get("nonce")
+    version = int(invoke_transaction["version"], 16)
+    nonce = invoke_transaction.get("nonce")
 
     common_data = {
-        "max_fee": int(request_body.get("max_fee", "0"), 16),
+        "max_fee": int(invoke_transaction.get("max_fee", "0"), 16),
         "version": version,
-        "signature": [int(data, 16) for data in request_body.get("signature", [])],
+        "signature": [
+            int(data, 16) for data in invoke_transaction.get("signature", [])
+        ],
         "nonce": int(nonce, 16) if nonce is not None else None,
     }
 
     if version == LEGACY_RPC_TX_VERSION:
         invoke_function = InvokeFunction(
-            contract_address=int(request_body["contract_address"], 16),
-            entry_point_selector=int(request_body["entry_point_selector"], 16),
-            calldata=[int(data, 16) for data in request_body.get("calldata", [])],
+            contract_address=int(invoke_transaction["contract_address"], 16),
+            entry_point_selector=int(invoke_transaction["entry_point_selector"], 16),
+            calldata=[int(data, 16) for data in invoke_transaction.get("calldata", [])],
             **common_data,
         )
     else:
         invoke_function = InvokeFunction(
-            contract_address=int(request_body["sender_address"], 16),
-            calldata=[int(data, 16) for data in request_body.get("calldata", [])],
+            contract_address=int(invoke_transaction["sender_address"], 16),
+            calldata=[int(data, 16) for data in invoke_transaction.get("calldata", [])],
             **common_data,
         )
 
     return invoke_function
 
 
+def make_declare(declare_transaction: RpcBroadcastedDeclareTxn) -> Declare:
+    """
+    Convert RpcDeclareTransaction to Declare
+    """
+    contract_class = declare_transaction["contract_class"]
+    if "abi" not in contract_class:
```

Comment:
When will this be the case?

Reply:
Abi is theoretically not required by the [specs](https://github.com/starkware-libs/starknet-specs/blob/v0.2.0/api/starknet_api_openrpc.json#L1656)